src/main/java/ai/ancf/lmos/operator/service/KubernetesResourceManager.java

```diff
@@ -0,0 +1,96 @@
+public class KubernetesResourceManager {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KubernetesResourceManager.class);
+
+    private final KubernetesClient kubernetesClient;
+
+    public KubernetesResourceManager(KubernetesClient kubernetesClient) {
+        this.kubernetesClient = kubernetesClient;
+    }
+
+    public void createOrUpdateAgentResource(ThingDescription thingDescription, Deployment deployment) {
+        AgentResource agentResource = new AgentResource();
+        agentResource.setMetadata(new ObjectMetaBuilder()
+            .withName(deployment.getMetadata().getName())
+            .withNamespace(deployment.getMetadata().getNamespace())
+            .build());
+
+        AgentSpec spec = new AgentSpec();
+        spec.setDescription(thingDescription.getDescription().get()
+                .toString());
+        spec.setThingDescription(thingDescription.toJsonString());
+
+        agentResource.setSpec(spec);
+
+        AgentResource agentResourceCreated = kubernetesClient.resources(AgentResource.class)
+                .inNamespace(deployment.getMetadata().getNamespace())
+                .withName(deployment.getMetadata().getName())
+                .createOrReplace(agentResource);
+
+        LOG.info("AgentResource {} created/updated for deployment: {}", agentResourceCreated.getFullResourceName(), deployment.getMetadata().getName());
+    }
+
+    public Service findService(Deployment deployment) {
+        Map<String, String> selectorLabels = deployment.getSpec().getSelector().getMatchLabels();
+        String deploymentNamespace = deployment.getMetadata().getNamespace();
+        ServiceList serviceList = kubernetesClient.services()
+                .inNamespace(deploymentNamespace)
+                .withLabels(selectorLabels)
+                .list();
+        if (serviceList.getItems().size() != 1) {
+            LOG.error("Expected exactly one service, but got {}, {}", serviceList.getItems().size(), serviceList.getItems());
+            throw new IllegalStateException("Expected exactly one service, but got " + serviceList.getItems().size());
+        }
+        return serviceList
+                .getItems()
+                .getFirst();
+    }
+
+    public String getBaseServiceUrl(Service service) {
+        ServicePort servicePort = service.getSpec().getPorts().getFirst();
+        String port = servicePort.getPort().toString();
+        boolean isHttps = servicePort.getPort() == 443 || "https".equalsIgnoreCase(servicePort.getName());
+        String protocol = isHttps ? "https://" : "http://";
+        String url = protocol + service.getMetadata().getName() + ":" + port;
+        LOG.info("Service URL is: {}", url);
+        return url;
+    }
+
+    public String getServiceUrl(Deployment deployment) {
+        String agentPath = deployment.getMetadata().getAnnotations().getOrDefault("wot/td-endpoint", null);
+        if(agentPath == null || agentPath.isEmpty()) {
```

Comment:
You call `getOrDefault`. Instead of null you can use `.well-known/wot`
You should verify if the path already starts with `/` and if yes you don't have to add it.

Reply:
resolved